### PR TITLE
Rename IoT

### PR DESCRIPTION
**Pull request by ctron**
### Type of change

- Refactoring

### Checklist

- [ ] Update/write design documentation in `./documentation/design`
- [ ] Write tests and make sure they pass
- [ ] Update documentation
- [ ] Check RBAC rights for Kubernetes / OpenShift roles
- [ ] Try your changes from Pod inside your Kubernetes and OpenShift cluster, not just locally
- [ ] Reference relevant issue(s) and close them after merging
- [ ] Update CHANGELOG.md

### CODE REVIEW

**Comment by ctron** [comment]
@enmasse-ci run tests ocp4 profile=iot

File: api-model/src/main/java/io/enmasse/iot/model/v1/IoTTenantSpec.java [hunk 1]
```diff
@@ -21,7 +21,7 @@
                 )
         )
 @JsonInclude(JsonInclude.Include.NON_NULL)
-public class IoTProjectSpec {
+public class IoTTenantSpec {
 
     private DownstreamStrategy downstream;
```
[the remark this answers is not in the document]
Comment (ctron) [reply api-model/src/main/java/io/enmasse/iot/model/v1/IoTTenantSpec.java:26]: :zombie: it's in zombie mode

**Comment by ctron** [comment]
@enmasse-ci run tests ocp4 profile=iot

**Comment by ctron** [comment]
@enmasse-ci run tests ocp4 profile=iot

**Comment by ctron** [comment]
@enmasse-ci run tests ocp4 profile=iot

File: iot/iot-device-registry-base/src/main/java/io/enmasse/iot/registry/tenant/TenantInformation.java [hunk 2]
```diff
@@ -34,15 +34,15 @@ public String getNamespace() {
...
+    public String getTenantName() {
+        return tenantName;
     }
 
     @Override
     protected ToStringHelper toStringHelper() {
         return super.toStringHelper()
                 .add("namespace", this.namespace)
-                .add("projectName", this.projectName)
+                .add("projectName", this.tenantName)
```
Comment (k-wall) [inline iot/iot-device-registry-base/src/main/java/io/enmasse/iot/registry/tenant/TenantInformation.java:45]:
```suggestion
                .add("tenantName", this.tenantName)
```

File: iot/iot-service-base/src/main/java/io/enmasse/iot/service/base/AbstractTenantBasedService.java [hunk 3]
```diff
@@ -129,28 +129,28 @@ protected void stopWatcher() {
...
+    protected Future<Optional<IoTTenant>> getTenant(final String tenantName) {
         return succeededFuture(
-                ofNullable(this.projects.get(tenantName))
-                        .map(project -> new IoTProjectBuilder(project).build()));
+                ofNullable(this.tenants.get(tenantName))
+                        .map(project -> new IoTTenantBuilder(project).build()));
     }
 
-    protected void onAdd(final IoTProject project) {
+    protected void onAdd(final IoTTenant project) {
```
Comment (k-wall) [inline iot/iot-service-base/src/main/java/io/enmasse/iot/service/base/AbstractTenantBasedService.java:138]: I'd be nice to rename these formal parameters.

File: iot/iot-tenant-service/src/main/java/io/enmasse/iot/tenant/impl/AbstractTenantService.java [hunk 4]
```diff
@@ -63,7 +63,7 @@ public void setTracer(Tracer tracer) {
         return get(subjectDn, NoopSpan.INSTANCE);
     }
 
-    protected TenantResult<JsonObject> convertToHono(final IoTProject project, final SpanContext spanContext) {
+    protected TenantResult<JsonObject> convertToHono(final IoTTenant project, final SpanContext spanContext) {
```
Comment (k-wall) [inline iot/iot-tenant-service/src/main/java/io/enmasse/iot/tenant/impl/AbstractTenantService.java:66]: likewise here

**Pull request merged**